src/Nethermind/Nethermind.BeaconNode.Storage/MemoryStoreProvider.cs

```diff
@@ -55,8 +56,13 @@ public MemoryStoreProvider(ILoggerFactory loggerFactory, IOptionsMonitor<TimePar
             return _store;
         }
 
-        public IStore? GetStore()
+        public IStore GetStore()
         {
+            if (_store is null)
+            {
+                throw new InvalidOperationException("Illegal attempt to retrieve a store that has not been created yet.");
```

Comment:
Sure. I think the API has some specific response codes for pre-genesis / while still syncing. But either works.